Client/WebFilters/ContentBlocker/BlocklistName.swift

```diff
@@ -88,36 +89,43 @@ class BlocklistName: CustomStringConvertible, ContentBlocker {
     return (onList, offList)
   }
 
-  static func compileBundledRules(ruleStore: WKContentRuleListStore) async {
-    await withTaskGroup(of: Void.self) { group in
-      BlocklistName.allLists.forEach { list in
-        group.addTask { await list.buildRule(ruleStore: ruleStore) }
-      }
-
-      // Compile block-cookie additionally
-      group.addTask { await BlocklistName.cookie.buildRule(ruleStore: ruleStore) }
-      await group.waitForAll()
-    }
+  static func compileBundledRules(ruleStore: WKContentRuleListStore) -> AnyPublisher<Void, Error> {
+    var allRules = BlocklistName.allLists.map({ $0.buildRule(ruleStore: ruleStore) })
+    
+    // Compile block-cookie additionally
+    allRules.append(BlocklistName.cookie.buildRule(ruleStore: ruleStore))
+    return Publishers.MergeMany(allRules)
+      .collect()
+      .flatMap({ $0.publisher })
+      .eraseToAnyPublisher()
   }
 
   func compile(
     data: Data?,
     ruleStore: WKContentRuleListStore = ContentBlockerHelper.ruleStore
-  ) async {
-
-    guard let data = data, let dataString = String(data: data, encoding: .utf8) else {
-      log.error("Could not read data for content blocker compilation.")
-      return
-    }
-
-    do {
-      let rule = try await ruleStore.compileContentRuleList(forIdentifier: filename, encodedContentRuleList: dataString)
+  ) -> AnyPublisher<Void, Error> {
+    return Future { [weak self] completion in
+      guard let self = self else {
+        completion(.failure("BlockListName Deallocated"))
```

Comment:
Should a completion block be called if the blocklist itself is no longer around?

Reply:
I think it should call it. I rather not have a dangling future :) but I did guard against the errors anyway. So if this does happen, it just logs the error to let us know why compilation failed so at least we'd know we need to hold a ref.